rdkit/Chem/TorsionFingerprints.py

```diff
@@ -479,6 +480,33 @@ def CalculateTFD(torsions1, torsions2, weights=None):
     tfd /= sum_weights
   return tfd
 
+def _getSameAtomOrder(mol1, mol2):
+  """ Generate a new molecule with the atom order of mol1 and coordinates
+      from mol2.
+      
+      Arguments:
+      - mol1:     first instance of the molecule of interest
+      - mol2:     second instance the molecule of interest
+
+      Return: RDKit molecule
+  """
+  match = mol2.GetSubstructMatch(mol1)
+  atomNums = tuple(range(mol1.GetNumAtoms()))
+  if match != atomNums: # atom orders are not the same!
+    #print "Atoms of second molecule reordered."
+    mol3 = Chem.Mol(mol1)
+    mol3.RemoveAllConformers()
+    for conf2 in mol2.GetConformers():
+      confId = conf2.GetId()
+      conf = rdchem.Conformer(mol1.GetNumAtoms())
+      conf.SetId(confId)
+      for i in range(mol1.GetNumAtoms()):
+        conf.SetAtomPosition(i, mol2.GetConformer(confId).GetAtomPosition(match[i]))
+      cid = mol3.AddConformer(conf)
+    return mol3
+  else:
+    return mol2
```

Comment:
Seems like you should be returning a copy of mol2 here (i.e. Chem.Mol(mol2)) in order to be consistent.